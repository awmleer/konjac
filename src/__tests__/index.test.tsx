import {act} from '@testing-library/react'
import * as React from 'react'
import {FC} from 'react'
import * as testing from '@testing-library/react'
import {KonjacProvider, useTranslation} from '../index'

test('can do translation', function () {
  const App: FC = (props) => {
    const {t} = useTranslation()
    return (
      <div>
        {t(['hello', '你好'])}
      </div>
    )
  }
  const renderer = testing.render(
    <KonjacProvider locale={'zh'} allLocales={['en', 'zh']}>
      <App/>
    </KonjacProvider>
  )
  expect(renderer.asFragment()).toMatchSnapshot()
})


test('empty messages', function () {
  const App: FC = (props) => {
    const {t} = useTranslation()
    let messages: string[]
    return (
      <div>
        {t(messages)}
      </div>
    )
  }
  const renderer = testing.render(
    <KonjacProvider locale={'zh'} allLocales={['en', 'zh']}>
      <App/>
    </KonjacProvider>
  )
  expect(renderer.asFragment()).toMatchSnapshot()
})


test('fallback', function () {
  const App: FC = (props) => {
    const {t} = useTranslation()
    return (
      <div>
        {t(['hello'])}
      </div>
    )
  }
  const renderer = testing.render(
    <KonjacProvider locale={'zh'} allLocales={['en', 'zh']} fallbackOrder={{'zh': 'en'}}>
      <App/>
    </KonjacProvider>
  )
  expect(renderer.asFragment()).toMatchSnapshot()
})

