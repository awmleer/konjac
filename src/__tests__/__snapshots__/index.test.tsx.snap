// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can do translation 1`] = `
<DocumentFragment>
  <div>
    你好
  </div>
</DocumentFragment>
`;

exports[`empty messages 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`fallback 1`] = `
<DocumentFragment>
  <div>
    hello
  </div>
</DocumentFragment>
`;
